client/network/src/protocol/notifications/behaviour.rs

```diff
@@ -1751,6 +1870,8 @@ impl NetworkBehaviour for Notifications {
 							trace!(target: "sub-libp2p", "External API <= Closed({}, {:?})", peer_id, set_id);
 							let event = NotificationsOut::CustomProtocolClosed { peer_id, set_id };
 							self.events.push_back(ToSwarm::GenerateEvent(event));
+							let _ = self.protocol_handles[usize::from(set_id)]
```

Comment:
I think the code starting from line 1839 handles the sink replacement if there is one available. Execution has reached this point (1873) because there is no sink available so I'm not sure what would the code give to the protocol handler since there is nothing to give.

Reply:
My comment is likely misleading — there is nothing interesting on line 1873, I just wasn't able to add a comment to lines 1846-1851 above because they are too far from the modified code, so ended up adding it here. When the flow reaches lines 1846-1851:
```rust
let event = NotificationsOut::CustomProtocolReplaced {
	peer_id,
	set_id,
	notifications_sink: replacement_sink,
};
self.events.push_back(ToSwarm::GenerateEvent(event));
```
we need to also replace the sink in the `NotificationsService` by calling `report_notification_sink_replaced()`, which is not done now.

Reply:
Yeah good catch. I don't know how I missed that lol